index.d.ts

```diff
@@ -0,0 +1,61 @@
+export type Replacer = (key: string, value: any) => void;
+export type SortKeys = (a: string, b: string) => number;
+
+export interface Options {
+	/**
+	 * Indentation as a string or number of spaces. Pass in null for no formatting.
+	 *
+	 * @default '\t'
+	 */
+	indent?: string | number | null;
+	/**
+	 * Detect indentation automatically if the file exists.
+	 *
+	 * @default false
+	 */
+	detectIndent?: boolean;
+	/**
+	 * Sort the keys recursively. Optionally pass in a compare function.
+	 *
+	 * @default false
+	 */
+	sortKeys?: SortKeys | boolean;
+	/**
+	 * Passed into JSON.stringify.
+	 */
+	replacer?: Replacer | Array<number | string>;
+	/**
+	 * Mode used when writing the file.
+	 *
+	 * @default 0o666
+	 */
+	mode?: number;
+}
+
+/**
+ * Stringify and write JSON to a file atomically.
+ *
+ * Creates directories for you as needed.
+ *
+ * @example
+ * import * as writeJsonFile from 'write-json-file';
+ *
+ * writeJsonFile.sync('foo.json', {foo: true});
+ * console.log('done');
+ */
+export function sync(filepath: string, data: object | number | string, options?: Options): void;
```

Comment:
What do you think about making `object | number | string` a named type called `JSONStringifyable`? I think that would make it slightly more readable as it would explain why it randomly only accepts object/number/string.

Reply:
Ok! I'll add it